internal/contour/listener.go

```diff
@@ -127,6 +139,68 @@ func (lvc *ListenerVisitorConfig) httpsAccessLog() string {
 	return DEFAULT_HTTPS_ACCESS_LOG
 }
 
+// accesslogType returns the access log type that should be configured
+// across all listener types or DEFAULT_ACCESS_LOG_TYPE if not configured.
+func (lvc *ListenerVisitorConfig) accesslogType() string {
+	if lvc.AccessLogType != "" {
+		return lvc.AccessLogType
+	}
+	return DEFAULT_ACCESS_LOG_TYPE
+}
+
+// accesslogFields returns the access log fields that should be configured
+// for Envoy, or a default set if not configured.
+func (lvc *ListenerVisitorConfig) accesslogFields() []string {
+	if lvc.AccessLogFields != nil {
+		return lvc.AccessLogFields
+	}
+	return []string{
+		"@timestamp",
+		"authority",
+		"bytes_received",
+		"bytes_sent",
+		"downstream_local_address",
+		"downstream_remote_address",
+		"duration",
+		"method",
+		"path",
+		"protocol",
+		"request_id",
+		"requested_server_name",
+		"response_code",
+		"response_flags",
+		"uber_trace_id",
+		"upstream_cluster",
+		"upstream_host",
+		"upstream_local_address",
+		"upstream_service_time",
+		"user_agent",
+		"x_forwarded_for",
+	}
+}
+
+func (lvc *ListenerVisitorConfig) newInsecureAccessLog() []*envoy_api_v2_accesslog.AccessLog {
+	var accesslogger []*envoy_api_v2_accesslog.AccessLog
+	switch lvc.accesslogType() {
+	case "json":
+		accesslogger = envoy.FileAccessLogJSON(lvc.httpAccessLog(), lvc.accesslogFields())
```

Comment:
case "json":
      return envoy.FileAccessLogJson...
default:
     return envoy.FileAccessLog...
}

Reply:
I wasn't sure if that style was okay, thanks.